File: orleans/GrainModels/Event.cs
using System;

namespace GrainModels
{
    public class Event
    {
        public string id { get; set; }
        public string title { get; set; }
        public string type { get; set; }
        public string start { get; set; }
        public string end { get; set; }
        public Topic[] topics { get; set; }
    }

    public class Topic
    {
        public string id { get; set; }
        public string desc { get; set; }        
    }
}

File: angular/src/app/models/event.ts
import { Topic } from './topic';

// Basic model to hold a event

export class Event {
  id: any;
  title: string;
  type: string;
  start: Date;
  end: Date;
  topics: Topic[];
}

var EventTypes = [ 'event', 'workshop', 'hack', 'lab' ];
export default EventTypes;